### PR TITLE
updated sailport.h

### PR DESCRIPTION
**Pull request by tarakreddy1984**
Fail over mode support

### CODE REVIEW

**Comment by rlhui** [comment]
Could we update the PR title to be more descriptive? Thanks.

**Comment by tarakreddy1984** [comment]
Fail over mode is ability to switch the primary port to secondary port and vice versa without loosing data when any fail fault occur on primary port.   This feature make sure continuous data transmission.

File: inc/saiport.h [hunk 1]
```diff
@@ -292,6 +292,57 @@ typedef enum _sai_port_prbs_config_t
...
+ * @brief Attribute data for #SAI_PORT_ATTR_FAIL_OVER_MODE
+ * Used for Fail over mode configuration on port
+ */
+typedef enum _sai_port_fail_over_mode_t
+{
+    /** Fail over mode disable */
+    SAI_PORT_FAIL_OVER_MODE_DISABLE,
+
+    /** Enable Fail over mode on the port */
+    SAI_PORT_FAIL_OVER_MODE_ENABLE,
```
Comment (rck-innovium) [inline inc/saiport.h:305]: The enum value SAI_PORT_FAIL_OVER_MODE_ENABLE seems redundant. If fail-over is enabled on one port, then it has to be either primary or secondary.

**Comment by lguohan** [comment]
this is fair fundamental change, @tarakreddy1984 ,  can you provide a design doc to describe the scenario here and also sample examples. you can refer to this pr. https://github.com/opencomputeproject/SAI/blob/master/doc/Hash/Policy-based-hash.md

**Comment by lguohan** [comment]
as we also discuess, such switchover is better fitted into port connector model design, can you enhance the port connector model in order to introduce this function.

**Review by tarakreddy1984 — COMMENTED** [review]
saiport.h updated

**Comment by tarakreddy1984** [comment]
Will create new request

**Pull request closed without merging**